### PR TITLE
ssh/tailssh: handle terminal opcodes

### PR DESCRIPTION
**Pull request by maisem**
Updates #3802

Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

**Comment by maisem** [comment]
Depends on https://github.com/maisem/ssh/commit/d40444fbdf12233013c0ef43a32dbf8d9d40fcd0

File: ssh/tailssh/incubator.go [hunk 1]
```diff
@@ -210,7 +274,7 @@ func startWithPTY(cmd *exec.Cmd, ptyReq ssh.Pty) (ptyFile *os.File, err error) {
 			tty.Close()
 		}
 	}()
-	ptyRawConn, err := ptyFile.SyscallConn()
+	ptyRawConn, err := tty.SyscallConn()
```
Comment (bradfitz) [inline ssh/tailssh/incubator.go:277]: whooops, I guess this mattered? did this fix something?

Comment (maisem) [reply ssh/tailssh/incubator.go:277]: nope :/

File: ssh/tailssh/incubator.go [hunk 2]
```diff
@@ -228,21 +292,29 @@ func startWithPTY(cmd *exec.Cmd, ptyReq ssh.Pty) (ptyFile *os.File, err error) {
...
+				tios.Ispeed = int(v)
+				continue
+			}
+			if c == gossh.TTY_OP_OSPEED {
+				tios.Ospeed = int(v)
+				continue
+			}
+			k, ok := terminalCodeMapping[c]
+			if !ok {
+				panic(fmt.Errorf("unknown opcode: %v", k))
```
Comment (bradfitz) [inline ssh/tailssh/incubator.go:306]: this is too aggressive

**Review by bradfitz — APPROVED** [review]
Remove the 42. Bump x ssh.

File: ssh/tailssh/incubator.go [hunk 3]
```diff
@@ -196,8 +197,78 @@ func resizeWindow(f *os.File, winCh <-chan ssh.Window) {
...
+	gossh.TTY_OP_ISPEED: "tty_op_ispeed",
+	gossh.TTY_OP_OSPEED: "tty_op_ospeed",
+
+	42: "iutf8", // defined in https://datatracker.ietf.org/doc/html/rfc8160.
+}
+
+// opcodeShortName returns the mnemonic of the opcode.
+// It returns "", false when not found.
+func opcodeShortName(opcode uint8) (m string, ok bool) {
+	m, ok = mnemonicMapping[opcode]
```
Comment (bradfitz) [inline ssh/tailssh/incubator.go:266]: This function doesn't really do anything? But sure?

**Review by bradfitz — COMMENTED** [review]
Can you remove the reflect bit now?

File: ssh/tailssh/incubator.go (hunk 3 above)
Comment (maisem) [reply ssh/tailssh/incubator.go]: removed, i wanted to move it to the ssh package initially. but can do that later.

**Pull request merged**